{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Available TSV files:\n",
      "1. Mitochondrial disorder with complex I deficiency.tsv\n",
      "2. Mitochondrial disorders (4).tsv\n",
      "3. Mitochondrial DNA maintenance disorder.tsv\n",
      "You selected: Mitochondrial disorder with complex I deficiency.tsv\n",
      "Available TXT files:\n",
      "1. IMDX6 filtered.txt\n",
      "You selected: IMDX6 filtered.txt\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "\n",
    "    \n",
    "# Get the .txt file to filter from\n",
    "# Get the list of files in the current working directory\n",
    "file_list2 = [file for file in os.listdir() if file.lower().endswith('.txt')]\n",
    "\n",
    "# Print the available CSV files for selection\n",
    "print(\"Available TXT files:\")\n",
    "for i, file_name in enumerate(file_list2):\n",
    "    print(f\"{i+1}. {file_name}\")\n",
    "\n",
    "# Prompt the user for input\n",
    "selection = input(\"Select a TXT file by entering the corresponding number: \")\n",
    "\n",
    "# Check if the user input is valid\n",
    "if selection.isdigit() and 1 <= int(selection) <= len(file_list2):\n",
    "    # Get the selected CSV file based on the user's input\n",
    "    selected_file2 = file_list2[int(selection) - 1]\n",
    "    print(f\"You selected: {selected_file2}\")\n",
    "else:\n",
    "    print(\"Invalid selection. Please try again.\")\n",
    "\n",
    "\n",
    "# Get the list of files in the current working directory\n",
    "file_list = [file for file in os.listdir() if file.lower().endswith('.tsv')]\n",
    "\n",
    "# Print the available CSV files for selection\n",
    "print(\"Available TSV files:\")\n",
    "for i, file_name in enumerate(file_list):\n",
    "    print(f\"{i+1}. {file_name}\")\n",
    "\n",
    "# Prompt the user for input\n",
    "selection = input(\"Select a CSV file by entering the corresponding number: \")\n",
    "\n",
    "# Check if the user input is valid\n",
    "if selection.isdigit() and 1 <= int(selection) <= len(file_list):\n",
    "    # Get the selected CSV file based on the user's input\n",
    "    selected_file = file_list[int(selection) - 1]\n",
    "    print(f\"You selected: {selected_file}\")\n",
    "else:\n",
    "    print(\"Invalid selection. Please try again.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The detected separator in the file is '\t'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\User\\AppData\\Local\\Temp\\ipykernel_1720\\4237271177.py:31: DtypeWarning: Columns (6,21) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(filepath2, sep=separator)\n"
     ]
    }
   ],
   "source": [
    "# Read the episode number file into a pandas dataframe\n",
    "\n",
    "# filepath = \"C:/Users/User/Documents/A-Flash-Disc/Data Extracts/Result_Listing_Extract_Vertical_41916_phumla.ndamane2023-03-15and1650.csv\"\n",
    "filepath = selected_file\n",
    "filepath = filepath.replace(\"\\\\\", \"/\")\n",
    "\n",
    "filepath2 = selected_file2\n",
    "filepath2 = filepath2.replace(\"\\\\\", \"/\")\n",
    "\n",
    "\n",
    "def detect_csv_separator(file_path2):\n",
    "    with open(file_path2, 'r', newline='') as file:\n",
    "        # Read a few lines from the file\n",
    "        sample_lines = [file.readline() for _ in range(0,2)]\n",
    "    # Define a list of possible separators to check\n",
    "    separators = [',', ';','|', '\\t']\n",
    "    # Iterate through each separator and count the occurrences in the sample lines\n",
    "    separator_counts = {}\n",
    "    for separator in separators:\n",
    "        counts = [line.count(separator) for line in sample_lines]\n",
    "        total_count = sum(counts)\n",
    "        separator_counts[separator] = total_count\n",
    "    # Determine the separator with the highest count\n",
    "    max_separator = max(separator_counts, key=separator_counts.get)\n",
    "    return max_separator\n",
    "\n",
    "csv_file = filepath2\n",
    "separator = detect_csv_separator(csv_file)\n",
    "print(f\"The detected separator in the file is '{separator}'\")\n",
    "\n",
    "df = pd.read_csv(filepath2, sep=separator)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>FREQS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Mastermind_MMID3</th>\n",
       "      <th>DisGeNET</th>\n",
       "      <th>GO</th>\n",
       "      <th>PHENOTYPES</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>.</td>\n",
       "      <td>1:69511-69511</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>79501</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_001005484.2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>1KG_ALL:G:NA</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>-</td>\n",
       "      <td>OR4F5:T141A,OR4F5:T162A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>.</td>\n",
       "      <td>1:924321-924321</td>\n",
       "      <td>G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>148398</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_001385640.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Retinitis_pigmentosa+DDG2P+ENSG00000187634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>1:924321-924321</td>\n",
       "      <td>G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>148398</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Retinitis_pigmentosa+DDG2P+ENSG00000187634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>1:924321-924321</td>\n",
       "      <td>G</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>148398</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_152486.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Retinitis_pigmentosa+DDG2P+ENSG00000187634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>1:924321-924321</td>\n",
       "      <td>G</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LINC02593</td>\n",
       "      <td>100130417</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NR_026874.2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Retinitis_pigmentosa+DDG2P+ENSG00000187634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317504</th>\n",
       "      <td>.</td>\n",
       "      <td>X:155774618-155774618</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>SPRY3</td>\n",
       "      <td>10251</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_001304990.2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2028</td>\n",
       "      <td>1KG_ALL:G:NA</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>-</td>\n",
       "      <td>SPRY3:Q249Q</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317505</th>\n",
       "      <td>.</td>\n",
       "      <td>X:155774618-155774618</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>SPRY3</td>\n",
       "      <td>10251</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_001394353.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2028</td>\n",
       "      <td>1KG_ALL:G:NA</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>-</td>\n",
       "      <td>SPRY3:Q249Q</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317506</th>\n",
       "      <td>.</td>\n",
       "      <td>X:155774618-155774618</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>SPRY3</td>\n",
       "      <td>10251</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_001394354.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2028</td>\n",
       "      <td>1KG_ALL:G:NA</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>-</td>\n",
       "      <td>SPRY3:Q249Q</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317507</th>\n",
       "      <td>.</td>\n",
       "      <td>X:155774618-155774618</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>SPRY3</td>\n",
       "      <td>10251</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_001394355.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2028</td>\n",
       "      <td>1KG_ALL:G:NA</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>-</td>\n",
       "      <td>SPRY3:Q249Q</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317508</th>\n",
       "      <td>.</td>\n",
       "      <td>X:155774618-155774618</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>SPRY3</td>\n",
       "      <td>10251</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_005840.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2028</td>\n",
       "      <td>1KG_ALL:G:NA</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>-</td>\n",
       "      <td>SPRY3:Q249Q</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>317509 rows Ã— 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Uploaded_variation               Location Allele  \\\n",
       "0                        .          1:69511-69511      G   \n",
       "1                        .        1:924321-924321      G   \n",
       "2                        .        1:924321-924321      G   \n",
       "3                        .        1:924321-924321      G   \n",
       "4                        .        1:924321-924321      G   \n",
       "...                    ...                    ...    ...   \n",
       "317504                   .  X:155774618-155774618      G   \n",
       "317505                   .  X:155774618-155774618      G   \n",
       "317506                   .  X:155774618-155774618      G   \n",
       "317507                   .  X:155774618-155774618      G   \n",
       "317508                   .  X:155774618-155774618      G   \n",
       "\n",
       "                  Consequence    IMPACT     SYMBOL       Gene Feature_type  \\\n",
       "0            missense_variant  MODERATE      OR4F5      79501   Transcript   \n",
       "1         5_prime_UTR_variant  MODIFIER     SAMD11     148398   Transcript   \n",
       "2         5_prime_UTR_variant  MODIFIER     SAMD11     148398   Transcript   \n",
       "3       upstream_gene_variant  MODIFIER     SAMD11     148398   Transcript   \n",
       "4       upstream_gene_variant  MODIFIER  LINC02593  100130417   Transcript   \n",
       "...                       ...       ...        ...        ...          ...   \n",
       "317504     synonymous_variant       LOW      SPRY3      10251   Transcript   \n",
       "317505     synonymous_variant       LOW      SPRY3      10251   Transcript   \n",
       "317506     synonymous_variant       LOW      SPRY3      10251   Transcript   \n",
       "317507     synonymous_variant       LOW      SPRY3      10251   Transcript   \n",
       "317508     synonymous_variant       LOW      SPRY3      10251   Transcript   \n",
       "\n",
       "               Feature         BIOTYPE  ... gnomADg_SAS_AF         FREQS  \\\n",
       "0       NM_001005484.2  protein_coding  ...         0.9772  1KG_ALL:G:NA   \n",
       "1       NM_001385640.1  protein_coding  ...         0.8798             -   \n",
       "2       NM_001385641.1  protein_coding  ...         0.8798             -   \n",
       "3          NM_152486.4  protein_coding  ...         0.8798             -   \n",
       "4          NR_026874.2          lncRNA  ...         0.8798             -   \n",
       "...                ...             ...  ...            ...           ...   \n",
       "317504  NM_001304990.2  protein_coding  ...         0.2028  1KG_ALL:G:NA   \n",
       "317505  NM_001394353.1  protein_coding  ...         0.2028  1KG_ALL:G:NA   \n",
       "317506  NM_001394354.1  protein_coding  ...         0.2028  1KG_ALL:G:NA   \n",
       "317507  NM_001394355.1  protein_coding  ...         0.2028  1KG_ALL:G:NA   \n",
       "317508     NM_005840.4  protein_coding  ...         0.2028  1KG_ALL:G:NA   \n",
       "\n",
       "       CLIN_SIG SOMATIC PHENO PUBMED         Mastermind_MMID3 DisGeNET GO  \\\n",
       "0             -     0,1   0,1      -  OR4F5:T141A,OR4F5:T162A        -  -   \n",
       "1             -       -     -      -                        -        -  -   \n",
       "2             -       -     -      -                        -        -  -   \n",
       "3             -       -     -      -                        -        -  -   \n",
       "4             -       -     -      -                        -        -  -   \n",
       "...         ...     ...   ...    ...                      ...      ... ..   \n",
       "317504        -     0,1   0,1      -              SPRY3:Q249Q        -  -   \n",
       "317505        -     0,1   0,1      -              SPRY3:Q249Q        -  -   \n",
       "317506        -     0,1   0,1      -              SPRY3:Q249Q        -  -   \n",
       "317507        -     0,1   0,1      -              SPRY3:Q249Q        -  -   \n",
       "317508        -     0,1   0,1      -              SPRY3:Q249Q        -  -   \n",
       "\n",
       "                                        PHENOTYPES  \n",
       "0                                                -  \n",
       "1       Retinitis_pigmentosa+DDG2P+ENSG00000187634  \n",
       "2       Retinitis_pigmentosa+DDG2P+ENSG00000187634  \n",
       "3       Retinitis_pigmentosa+DDG2P+ENSG00000187634  \n",
       "4       Retinitis_pigmentosa+DDG2P+ENSG00000187634  \n",
       "...                                            ...  \n",
       "317504                                           -  \n",
       "317505                                           -  \n",
       "317506                                           -  \n",
       "317507                                           -  \n",
       "317508                                           -  \n",
       "\n",
       "[317509 rows x 73 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The detected separator in the file is '\t'\n"
     ]
    }
   ],
   "source": [
    "# Read the filter list in by using the read_csv function in pandas\n",
    "# filepath = r\"C:\\Users\\User\\Documents\\A-Flash-Disc\\Bioinformatics\\Mitochondrial disorder with complex I deficiency.tsv\"\n",
    "csv_file = filepath\n",
    "separator = detect_csv_separator(csv_file)\n",
    "print(f\"The detected separator in the file is '{separator}'\")\n",
    "\n",
    "df_filtering = pd.read_csv(filepath, sep=separator)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Entity Name Entity type Gene Symbol         Sources(; separated)  \\\n",
      "0       ACAD9        gene       ACAD9  Expert Review Green;NHS GMS   \n",
      "1     FOXRED1        gene     FOXRED1  Expert Review Green;NHS GMS   \n",
      "2      NDUFA1        gene      NDUFA1  Expert Review Green;NHS GMS   \n",
      "3     NDUFA10        gene     NDUFA10  Expert Review Green;NHS GMS   \n",
      "4     NDUFA11        gene     NDUFA11  Expert Review Green;NHS GMS   \n",
      "\n",
      "                                             Level4  Level3  Level2  \\\n",
      "0  Mitochondrial disorder with complex I deficiency     NaN     NaN   \n",
      "1  Mitochondrial disorder with complex I deficiency     NaN     NaN   \n",
      "2  Mitochondrial disorder with complex I deficiency     NaN     NaN   \n",
      "3  Mitochondrial disorder with complex I deficiency     NaN     NaN   \n",
      "4  Mitochondrial disorder with complex I deficiency     NaN     NaN   \n",
      "\n",
      "                                Model_Of_Inheritance  \\\n",
      "0            BIALLELIC, autosomal or pseudoautosomal   \n",
      "1            BIALLELIC, autosomal or pseudoautosomal   \n",
      "2  X-LINKED: hemizygous mutation in males, monoal...   \n",
      "3            BIALLELIC, autosomal or pseudoautosomal   \n",
      "4            BIALLELIC, autosomal or pseudoautosomal   \n",
      "\n",
      "                                          Phenotypes  Omim  ...  \\\n",
      "0  Mitochondrial complex I deficiency, nuclear ty...   NaN  ...   \n",
      "1  Mitochondrial complex I deficiency, nuclear ty...   NaN  ...   \n",
      "2  Mitochondrial complex I deficiency, nuclear ty...   NaN  ...   \n",
      "3  Mitochondrial complex I deficiency, nuclear ty...   NaN  ...   \n",
      "4  Mitochondrial complex I deficiency, nuclear ty...   NaN  ...   \n",
      "\n",
      "   Position GRCh38 Start  Position GRCh38 End STR Repeated Sequence  \\\n",
      "0                    NaN                  NaN                   NaN   \n",
      "1                    NaN                  NaN                   NaN   \n",
      "2                    NaN                  NaN                   NaN   \n",
      "3                    NaN                  NaN                   NaN   \n",
      "4                    NaN                  NaN                   NaN   \n",
      "\n",
      "   STR Normal Repeats  STR Pathogenic Repeats  \\\n",
      "0                 NaN                     NaN   \n",
      "1                 NaN                     NaN   \n",
      "2                 NaN                     NaN   \n",
      "3                 NaN                     NaN   \n",
      "4                 NaN                     NaN   \n",
      "\n",
      "   Region Haploinsufficiency Score Region Triplosensitivity Score  \\\n",
      "0                              NaN                            NaN   \n",
      "1                              NaN                            NaN   \n",
      "2                              NaN                            NaN   \n",
      "3                              NaN                            NaN   \n",
      "4                              NaN                            NaN   \n",
      "\n",
      "   Region Required Overlap Percentage  Region Variant Type  \\\n",
      "0                                 NaN                  NaN   \n",
      "1                                 NaN                  NaN   \n",
      "2                                 NaN                  NaN   \n",
      "3                                 NaN                  NaN   \n",
      "4                                 NaN                  NaN   \n",
      "\n",
      "   Region Verbose Name  \n",
      "0                  NaN  \n",
      "1                  NaN  \n",
      "2                  NaN  \n",
      "3                  NaN  \n",
      "4                  NaN  \n",
      "\n",
      "[5 rows x 36 columns]\n"
     ]
    }
   ],
   "source": [
    "print(df_filtering.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       #Uploaded_variation               Location Allele  \\\n",
      "19820                    .  1:161193247-161193247      C   \n",
      "19821                    .  1:161193247-161193247      C   \n",
      "19824                    .  1:161198214-161198214      T   \n",
      "19826                    .  1:161198214-161198214      T   \n",
      "19827                    .  1:161198214-161198214      T   \n",
      "...                    ...                    ...    ...   \n",
      "261770                   .    6:96891202-96891202      G   \n",
      "261771                   .    6:96891212-96891212      T   \n",
      "298742                   .    9:32552464-32552464      G   \n",
      "298746                   .    9:32552464-32552464      G   \n",
      "298748                   .    9:32552464-32552464      G   \n",
      "\n",
      "                    Consequence    IMPACT   SYMBOL   Gene Feature_type  \\\n",
      "19820     upstream_gene_variant  MODIFIER   NDUFS2   4720   Transcript   \n",
      "19821     upstream_gene_variant  MODIFIER   NDUFS2   4720   Transcript   \n",
      "19824     upstream_gene_variant  MODIFIER   NDUFS2   4720   Transcript   \n",
      "19826            intron_variant  MODIFIER   NDUFS2   4720   Transcript   \n",
      "19827     upstream_gene_variant  MODIFIER   NDUFS2   4720   Transcript   \n",
      "...                         ...       ...      ...    ...          ...   \n",
      "261770       synonymous_variant       LOW  NDUFAF4  29078   Transcript   \n",
      "261771       synonymous_variant       LOW  NDUFAF4  29078   Transcript   \n",
      "298742  downstream_gene_variant  MODIFIER   NDUFB6   4712   Transcript   \n",
      "298746  downstream_gene_variant  MODIFIER   NDUFB6   4712   Transcript   \n",
      "298748  downstream_gene_variant  MODIFIER   NDUFB6   4712   Transcript   \n",
      "\n",
      "               Feature         BIOTYPE  ... gnomADg_SAS_AF         FREQS  \\\n",
      "19820   NM_001377298.1  protein_coding  ...         0.7495  1KG_ALL:C:NA   \n",
      "19821   NM_001377300.1  protein_coding  ...         0.7495  1KG_ALL:C:NA   \n",
      "19824   NM_001166159.2  protein_coding  ...         0.3843  1KG_ALL:T:NA   \n",
      "19826   NM_001377298.1  protein_coding  ...         0.3843  1KG_ALL:T:NA   \n",
      "19827   NM_001377299.1  protein_coding  ...         0.3843  1KG_ALL:T:NA   \n",
      "...                ...             ...  ...            ...           ...   \n",
      "261770     NM_014165.4  protein_coding  ...         0.5037  1KG_ALL:G:NA   \n",
      "261771     NM_014165.4  protein_coding  ...         0.9094             -   \n",
      "298742  NM_001199987.2  protein_coding  ...        0.02132  1KG_ALL:G:NA   \n",
      "298746     NM_002493.5  protein_coding  ...        0.02132  1KG_ALL:G:NA   \n",
      "298748     NM_182739.3  protein_coding  ...        0.02132  1KG_ALL:G:NA   \n",
      "\n",
      "                    CLIN_SIG SOMATIC PHENO  \\\n",
      "19820                      -     0,1   1,1   \n",
      "19821                      -     0,1   1,1   \n",
      "19824                      -     0,1   1,1   \n",
      "19826                      -     0,1   1,1   \n",
      "19827                      -     0,1   1,1   \n",
      "...                      ...     ...   ...   \n",
      "261770  benign,likely_benign     0,1   1,1   \n",
      "261771  benign,likely_benign       -     1   \n",
      "298742                benign     0,1   1,1   \n",
      "298746                benign     0,1   1,1   \n",
      "298748                benign     0,1   1,1   \n",
      "\n",
      "                                                   PUBMED  \\\n",
      "19820   27917384,19064571,26495885,28081267,28742099,3...   \n",
      "19821   27917384,19064571,26495885,28081267,28742099,3...   \n",
      "19824        22545919,28081267,28742099,31361310,32929287   \n",
      "19826        22545919,28081267,28742099,31361310,32929287   \n",
      "19827        22545919,28081267,28742099,31361310,32929287   \n",
      "...                                                   ...   \n",
      "261770                                                  -   \n",
      "261771                                  27029813,20029939   \n",
      "298742                                                  -   \n",
      "298746                                                  -   \n",
      "298748                                                  -   \n",
      "\n",
      "                                         Mastermind_MMID3  \\\n",
      "19820                                                   -   \n",
      "19821                                                   -   \n",
      "19824   ADAMTS4:S138S,NDUFS2:5UTR,NDUFS2:5UTRint,NDUFS...   \n",
      "19826   ADAMTS4:S138S,NDUFS2:5UTR,NDUFS2:5UTRint,NDUFS...   \n",
      "19827   ADAMTS4:S138S,NDUFS2:5UTR,NDUFS2:5UTRint,NDUFS...   \n",
      "...                                                   ...   \n",
      "261770                                      NDUFAF4:L144L   \n",
      "261771                                      NDUFAF4:Q140Q   \n",
      "298742                                                  -   \n",
      "298746                                                  -   \n",
      "298748                                                  -   \n",
      "\n",
      "                                                 DisGeNET GO  \\\n",
      "19820   27706574:0.01:Degenerative_polyarthritis,26495...  -   \n",
      "19821   27706574:0.01:Degenerative_polyarthritis,26495...  -   \n",
      "19824                                                   -  -   \n",
      "19826                                                   -  -   \n",
      "19827                                                   -  -   \n",
      "...                                                   ... ..   \n",
      "261770                                                  -  -   \n",
      "261771                                                  -  -   \n",
      "298742                                                  -  -   \n",
      "298746                                                  -  -   \n",
      "298748                                                  -  -   \n",
      "\n",
      "                                               PHENOTYPES  \n",
      "19820   HDL_cholesterol_levels+NHGRI-EBI_GWAS_catalog+...  \n",
      "19821   HDL_cholesterol_levels+NHGRI-EBI_GWAS_catalog+...  \n",
      "19824   Mitochondrial_complex_I_deficiency+DDG2P+ENSG0...  \n",
      "19826   Mitochondrial_complex_I_deficiency+DDG2P+ENSG0...  \n",
      "19827   Mitochondrial_complex_I_deficiency+DDG2P+ENSG0...  \n",
      "...                                                   ...  \n",
      "261770  MITOCHONDRIAL_COMPLEX_I_DEFICIENCY__NUCLEAR_TY...  \n",
      "261771  ClinVar:_phenotype_not_specified+ClinVar+rs114...  \n",
      "298742  RETINITIS_PIGMENTOSA_31+DDG2P+ENSG00000197579,...  \n",
      "298746  RETINITIS_PIGMENTOSA_31+DDG2P+ENSG00000197579,...  \n",
      "298748  RETINITIS_PIGMENTOSA_31+DDG2P+ENSG00000197579,...  \n",
      "\n",
      "[215 rows x 73 columns]\n"
     ]
    }
   ],
   "source": [
    "# Filter the df dataframe \"SYMBOL\" column using the df_filter \"Gene Symbol\" column\n",
    "filter_list = df_filtering[\"Gene Symbol\"].tolist()\n",
    "\n",
    "df_filtered = df[df[\"SYMBOL\"].isin(filter_list)]\n",
    "print(df_filtered)\n",
    "\n",
    "# merge the df_filtering[\"Sources(; seperated)\"] column with the df_filtered dataframe\n",
    "df_full_dataset = pd.merge(df_filtered, df_filtering, left_on=\"SYMBOL\", right_on=\"Gene Symbol\")\n",
    "\n",
    "# send the df_filtered dataframe to a csv file\n",
    "df_full_dataset.to_csv(f\"{selected_file2}_and_{selected_file}.csv\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
